docs/dev/storage-engine.md

```diff
@@ -0,0 +1,16 @@
+## Storage Engine
+The storage engine component is responsible for handling data points while converting them from their initial uncompressed 
+state to their final compressed model-based state. Since data is compressed in segments, the storage engine is responsible 
+for efficiently managing these segments until they can be compressed. When compressed, the storage engine manages the 
+data until it can be saved to disk. The main objective is to avoid unnecessary I/O by having an efficient in-memory data 
```

Comment:
Maybe replace `The main objective is` with `The storage engine is designed`.

Reply:
Fixed.